app/utils/reducerInjectors.js

```diff
@@ -0,0 +1,57 @@
+import invariant from 'invariant';
+import isEmpty from 'lodash/isEmpty';
+import isFunction from 'lodash/isFunction';
+import isString from 'lodash/isString';
+
+import checkStore from './checkStore';
+import createReducer from '../reducers';
+
+const allowedModes = ['restart-on-remount', 'daemon'];
+
+export function injectReducerFactory(store, isValid) {
+  return function injectReducer(name, reducer) {
+    if (!isValid) checkStore(store);
+
+    invariant(
+      isString(name) && !isEmpty(name) && isFunction(reducer),
+      '(app/utils...) injectReducer: Expected `reducer` to be a reducer function'
+    );
+
+    // Check `store.injectedReducers[name] === reducer` for hot reloading when a name is the same but a reducer is different
+    // otherwise daemon reducers won't hot reload
+    if (Reflect.has(store.injectedReducers, name) && store.injectedReducers[name] === reducer) return;
+
+    store.injectedReducers[name] = reducer; // eslint-disable-line no-param-reassign
+    store.replaceReducer(createReducer(store.injectedReducers));
+  };
+}
+
+export function ejectReducerFactory(store, isValid) {
+  return function ejectReducer(name, mode = 'restart-on-remount') {
+    if (!isValid) checkStore(store);
+
+    invariant(
+      isString(name) && !isEmpty(name),
+      '(app/utils...) injectReducer: Expected `name` and `mode` to be a non empty string with a correct value'
+    );
+
+    invariant(
+      isString(mode) && allowedModes.includes(mode),
+      `(app/utils...) injectReducer: Expected \`mode\` to be one of: ${allowedModes.join(', ')}`
+    );
+
+    if (mode !== 'daemon') {
+      Reflect.deleteProperty(store.injectedReducers, name);
+      store.replaceReducer(createReducer(store.injectedReducers));
+    }
+  };
+}
+
+export default function getInjectors(store) {
```

Comment:
> do you mean to hide those injectors in the file with a decorator and make them private? And rename the file to withInjectors?

hmm... actually i initially was thinking about an HOC specifically [for the use-case you described](https://github.com/anuraaga/react-boilerplate/pull/4#discussion_r125732318) so we could handle composition ie. with connect, but thinking more about it, it may not be a bad idea to pass saga and reducer injectors in a single decorator... similar api to the apollo example or react-redux's connect which accepts objects for mapState, mapDispatch etc. in theory we could accept mapState, mapDispatch etc. and abstract the need for users to access store or `compose` with connect... basically:

```js
export const SettingsRoute = withConnectedBundle({
  entry: Loadable, 
  inject: { reducer, saga },
  mapStateToProps, // pass thru
  mapDispatchToProps
})
```

let me know what you think. side note: i have been running some webpack analysis on the new code, and split points are perfect 😄